Modernize documentation

jyang81 opened the pull request:
# Alaska Airlines Pull Request

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

## Reviewer Checklist:

#### Documentation Accuracy
- [x] Do all explanations in the API table convey the feature they are referencing in an understandable way?
    - [x] Are all `String` properties displaying all of their "options"?
- [x] Are all necessary API features documented? Are there any missing that you believe should be added?
- [x] Does the component description and use case section convey what the component is all about? Is there too much or too little information?
- [x] Ensure only NECESSARY examples are on the `index` page (i.e. only the most "basic" example(s) of component)
- [ ] Are the API examples more or less appearance in the same order as they are in on the API table?
- [ ] Ensure custom registration example is now on `install` page and renders as expected
- [x] Do all functional JS examples still work?

#### Code Changes
- [x] Ensure all example files are rename from camel case to kebab case (ex. `myExample.html` -> `my-example.html`)
- [x] Ensure no backticks appear in example header names
- [x] Is there a `perf` commit to trigger a release, as expected?

#### Visual Checks
- [x] Ensure all `See Code` accordions actually render the code blocks under their respective examples
- [x] Do all examples with multiple components in them have sufficient spacing?

**By submitting this Pull Request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

_Pull Requests will be evaluated by their quality of update and whether it is consistent with the goals and values of this project. Any submission is to be considered a conversation between the submitter and the maintainers of this project and may require changes to your submission._

**Thank you for your submission!**<br>
-- Auro Design System Team

## Summary by Sourcery

Modernize and reorganize auro-background documentation and examples while aligning API docs with the current component surface and metadata.

Enhancements:
- Clarify and restructure README and docs to focus on use cases, basic examples, and custom component registration for version management.
- Update API documentation to include typed properties, attributes, inset value options, slots, and the register method.
- Improve inline component JSDoc/property comments for better generated documentation and custom-elements.json output.
- Refresh example snippets to use current auro-header and auro-hyperlink APIs and updated breakpoint/inset values.

Build:
- Bump lit and Auro-related dev dependencies to newer versions and expose custom-elements.json via exports and the customElements field in package.json.

jyang81 commented on docs/api.md:
```diff
@@ -2,23 +2,35 @@
+
+| Name     | Parameters                                                          | Return | Description                                       |
+| -------- | ------------------------------------------------------------------- | ------ | ------------------------------------------------- |
+| register | `name` (string) - The name of element that you want to register to. |        | This will register this element with the browser. |
```
fixed

jordanjones243 commented on README.md:
```diff
@@ -28,17 +28,21 @@ The `auro-background` element is an easy-to-use mobile-first alternative to writ
 
-<!-- AURO-GENERATED-CONTENT:START (REMOTE:url=https://raw.githubusercontent.com/AlaskaAirlines/auro-templates/main/templates/default/partials/browserSupport.md) -->
-For the most up to date information on [UI development browser support](https://auro.alaskaair.com/support/browsersSupport)
+<!-- AURO-GENERATED-CONTENT:START (FILE:src=./docs/partials/useCases.md) -->
```
This "Use Cases" section needs to be updated --- This is not actually describing the use case for this component

jordanjones243 replied:
If you think that info is valuable, it could be added to the `readmeAddlInfo.md` file

jordanjones243 commented on apiExamples/breakpoints.html:
```diff
@@ -1,10 +1,10 @@
-    <auro-header level="2" display="600">View all destinations</auro-header>
-    <auro-hyperlink cta href="#">See where we fly</auro-hyperlink>
+    <auro-header level="2" visualLevel="3">View all destinations</auro-header>
+    <auro-hyperlink type="cta" href="#">See where we fly</auro-hyperlink>
```
I think we should be using `appearance="inverse"` rather than `type="cta"` for all examples. Not a big fan of how the button looks in the background used for examples.

jordanjones243 commented on docTemplates/README.md:
```diff
@@ -25,61 +25,58 @@ The following sections are editable by making changes to the following files:
 <!-- AURO-GENERATED-CONTENT:START (FILE:src=./docs/partials/readmeAddlInfo.md) -->
```
This should be in the `.gitignore` and should not be committed into repo

jordanjones243 commented on docs/partials/index.md:
```diff
@@ -1,55 +1,33 @@
-# Background
-
-<!-- AURO-GENERATED-CONTENT:START (FILE:src=../docs/partials/description.md) -->
+# {{ capitalize name }}
```
Although I think this works, there is no need to use the handlebars syntax

jordanjones243 commented on src/auro-background.js:
```diff
@@ -49,21 +34,73 @@ export class AuroBackground extends LitElement {
+       */
       bg: { type: String },
+      /**
+       * Sets the background image at `breakpoint-sm` and above
```
We need to specify what `breakpoint-*` actually represents. I don't think that is exactly what we call it in Auro. This comment applies to all API property descriptions below.

jordanjones243 commented on src/auro-background.js:
```diff
@@ -49,21 +34,73 @@ export class AuroBackground extends LitElement {
       widthLg: { type: String },
+      /**
+       * Applies internal padding at all breakpoints
+       * @type {'none' | 'xxxs' | 'xxs' | 'xs' | 'sm' | 'md' | 'lg' | 'xl' | 'xxl' | 'xxxl' | string}
```
Remove `string` from the list here. Applies to all properties using the `@type` idenitifier.

jordanjones243 reviewed (dismissed):
Make sure to update the ticket number in the initial commit of this PR

jyang81 commented:
@jordanjones243 is taking over this PR

jordanjones243 committed:
perf: update and modernize component documentation #72
example code updated to reflect latest Auro element APIs
example documentation layout improved
custom registration updated and moved to install page
core element descriptions updated
properties now listed alphabetically
doc build system now uses the latest version of auro-cli
package json configuration updated to support latest auro-cli
additional packages dependencies updated to latest versions

chrisfalaska commented on docs/partials/api.md:
```diff
@@ -1,85 +1,90 @@
+Use the `inset` attribute to add padding inside the element.
 
-You may specify different `bg` values for each breakpoint using the `bgSm`, `bgMd` and `bgLg` properties.
+The `inset` attribute supports the following values: 'none', 'xxxs', 'xxs', 'xs', 'sm', 'md', 'lg', 'xl', 'xxl', 'xxxl' or any CSS value which is valid for the CSS `padding` rule (e.g. '1rem').
```
These should use backticks ``

chrisfalaska commented on docs/partials/api.md:
```diff
@@ -1,85 +1,90 @@
-You may specify different `bg` values for each breakpoint using the `bgSm`, `bgMd` and `bgLg` properties.
+The `inset` attribute supports the following values: 'none', 'xxxs', 'xxs', 'xs', 'sm', 'md', 'lg', 'xl', 'xxl', 'xxxl' or any CSS value which is valid for the CSS `padding` rule (e.g. '1rem').
+
+Note: When using a custom CSS rule do not include a semi-colon at at the end of the string (e.g. '1rem' is valid, '1rem;' is invalid).
```
Same here

chrisfalaska commented on docs/partials/api.md:
```diff
@@ -1,85 +1,90 @@
+Use the `height` and `width` attributes to set a fixed height/width on the element. 
 
-#### Use fixed `height` and `width`
+The `height` and `width` attributes accept any valid CSS value for height/width (e.g. '100px', '12rem', '50vw').
```
These values should be in backticks

chrisfalaska commented:
The API table is arranged:

- Background
- Height
- Inset
- Width

However, the examples below the table do not follow this order.

chrisfalaska commented:
I do not see an Install page:

http://localhost:8001/install

jordanjones243 commented:
> The API table is arranged:
> 
> * Background
> * Height
> * Inset
> * Width
> 
> However, the examples below the table do not follow this order.

I've edited this, but height/width are combined in a single example so they might appear a little different than the order of the API table.

jordanjones243 commented:
> I do not see an Install page:
> 
> http://localhost:8001/install

There is no `/install` endpoint because the content is coming from the README. In order to check this content, you need to copy the root `README.md` and paste that into the `/demo` folder. You will then navigate to the `demo/index.html` file and change line 43 from `fetch('./index.md')` to `fetch('./README.md')`.
